### ISSUE TITLE
Error encountered after CRA build: cannot convert a BigInt value to a number.

### ISSUE THREAD

**Issue by immaxiao** [polkadot-js/api#5640]
hi,

I use create react app to create a project, only add @polkadot/api and import it, it can run correctly in dev server.

After I build the production package, when opened in a browser there is an error: cannot convert a BigInt value to a number.
<img width="1061" alt="image" src="https://user-images.githubusercontent.com/116542616/237061761-214be30f-202b-4b5a-bd5a-b20ef76871c4.png">

I only add `@polkadot/api` and create api as follows:
```
import logo from './logo.svg';
import './App.css';
import { useEffect } from 'react';
import { ApiPromise, WsProvider } from '@polkadot/api';

function App() {
  useEffect(() => {
    const wsProvider = new WsProvider('wss://rpc.polkadot.io');
    ApiPromise.create({ provider: wsProvider });
  }, [])
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </header>
    </div>
  );
}

export default App;
```
The node version I am using is v16.18.0, and the package.json is as follows
```
{
  "name": "my-app",
  "version": "0.1.0",
  "private": true,
  "homepage": ".",
  "dependencies": {
    "@polkadot/api": "^10.6.1",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.4.0",
    "@testing-library/user-event": "^13.5.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-scripts": "5.0.1",
    "web-vitals": "^2.1.4"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  }
}
```

I also used Nextjs for testing, but I didn't find this problem. 

Thanks.

* **I'm submitting a ...**

  - [x] Bug report
  - [ ] Feature request
  - [ ] Support request
  - [ ] Other

* **What is the current behavior and expected behavior?**

* **What is the motivation for changing the behavior?**

* **Please tell us about your environment:**

  - Version:
  - Environment:

    - [ ] Node.js
    - [x] Browser
    - [ ] Other (limited support for other environments)

  - Language:

    - [ ] JavaScript
    - [x] TypeScript (include tsc --version)
    - [ ] Other

**Comment by jacogr** [comment]
It is your build - basically your babel config changes `10n ^ 2n` into `Math.pow(10n, 2n)` into the compiled output, which is invalid.

There are issues logged (search here), for solutions to change your config.

TL;DR It is a known Babel issue, it is not API related.

**Comment by jacogr** [comment]
Some example from around the web -

https://github.com/paulmillr/noble-ed25519/issues/23
https://github.com/paulmillr/noble-secp256k1/issues/31
https://forum.dfinity.org/t/cannot-convert-a-bigint-value-to-a-number/5470
https://github.com/polkadot-js/common/issues/1573

**Comment by jacogr** [comment]
Also covered in the build FAQ, see https://polkadot.js.org/docs/usage/FAQ/#under-my-babel-build-i-have-a-bigint-to-number-conversion-error

**Comment by jacogr** [comment]
As per the last issue, there is also an entry in the build FAQ that may be helpful - https://polkadot.js.org/docs/usage/FAQ/#under-my-babel-build-i-have-a-bigint-to-number-conversion-error

**Comment by immaxiao** [comment]
Thanks for the information you provided, it works well. This issue has been solved.
